lib/util.js

```diff
@@ -63,11 +67,12 @@ schemaFaker.option({
 * @param {*} components list of predefined components (with schemas)
 * @param {string} schemaFormat default or xml
 * @param {string} indentCharacter char for 1 unit of indentation
+* @param {string} bodyType tells that the schema object is of request or response
 * @returns {object} fakedObject
 */
-function safeSchemaFaker(oldSchema, components, schemaFormat, indentCharacter) {
+function safeSchemaFaker(oldSchema, components, schemaFormat, indentCharacter, bodyType) {
   var prop,
-    schema = deref.resolveRefs(oldSchema, components);
+    schema = deref.resolveRefs(oldSchema, bodyType, components);
```

Comment:
bodyType is added b/w schema and components here. As far as possible, we should maintain the same order of args for `safeSchemaFaker` too